### PR TITLE
Restaurant/feature/add update restuarant admin

### PR DESCRIPTION
**Pull request by NadaFeteiha**

### CODE REVIEW

**Review by mosayed01 — APPROVED** [review]

**Review by Asiasama710 — APPROVED** [review]

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt [hunk 1]
```diff
@@ -34,18 +34,16 @@ fun Route.restaurantRoutes() {
     route("/restaurant") {
 
         get("/{id}") {
-            val restaurantId = call.parameters["id"] ?: throw MissingParameterException(
-                INVALID_REQUEST_PARAMETER
-            )
+            val restaurantId = call.parameters["id"] ?:
+            throw MultiErrorException(listOf( INVALID_REQUEST_PARAMETER))
```
Comment (mustafa-sab3alarab) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt:38]:
```suggestion
            throw MultiErrorException(listOf(INVALID_REQUEST_PARAMETER))
```

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt [hunk 2]
```diff
@@ -34,18 +34,16 @@ fun Route.restaurantRoutes() {
...
             call.respond(HttpStatusCode.OK, restaurant)
         }
 
         get("/{id}/categories") {
             val restaurantId =
-                call.parameters.extractString("id") ?: throw MissingParameterException(
-                    INVALID_REQUEST_PARAMETER
-                )
+                call.parameters.extractString("id") ?:
+                throw MultiErrorException(listOf( INVALID_REQUEST_PARAMETER))
```
Comment (mustafa-sab3alarab) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt:46]:
```suggestion
                throw MultiErrorException(listOf(INVALID_REQUEST_PARAMETER))
```

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt [hunk 3]
```diff
@@ -58,34 +56,38 @@ fun Route.restaurantRoutes() {
 
         post("/{id}/categories") {
             val restaurantId =
-                call.parameters.extractString("id") ?: throw MissingParameterException(
-                    INVALID_REQUEST_PARAMETER
-                )
+                call.parameters.extractString("id") ?:
+                throw MultiErrorException(listOf( INVALID_REQUEST_PARAMETER))
```
Comment (mustafa-sab3alarab) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt:60]:
```suggestion
                throw MultiErrorException(listOf(INVALID_REQUEST_PARAMETER))
```

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt [hunk 4]
```diff
@@ -58,34 +56,38 @@ fun Route.restaurantRoutes() {
...
+            call.respond(HttpStatusCode.OK, result)
+        }
+
         delete("/{id}/categories") {
             val restaurantId =
-                call.parameters.extractString("id") ?: throw MissingParameterException(
-                    INVALID_REQUEST_PARAMETER
-                )
+                call.parameters.extractString("id") ?:
+                throw MultiErrorException(listOf( INVALID_REQUEST_PARAMETER))
```
Comment (mustafa-sab3alarab) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt:81]:
```suggestion
                throw MultiErrorException(listOf(INVALID_REQUEST_PARAMETER))
```

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt [hunk 5]
```diff
@@ -58,34 +56,38 @@ fun Route.restaurantRoutes() {
...
+                manageRestaurantDetails.deleteCategoriesInRestaurant(restaurantId, categoryIds)
             call.respond(HttpStatusCode.OK, result)
         }
 
         delete("/{id}") {
-            val restaurantId = call.parameters["id"] ?: throw MissingParameterException(
-                INVALID_REQUEST_PARAMETER
-            )
+            val restaurantId = call.parameters["id"] ?:
+            throw MultiErrorException(listOf( INVALID_REQUEST_PARAMETER))
```
Comment (mustafa-sab3alarab) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/RestaurantRoutes.kt:90]:
```suggestion
            throw MultiErrorException(listOf(INVALID_REQUEST_PARAMETER))
```

File: service_restaurant/src/test/kotlin/org/thechance/service_restaurant/domain/validation/RestaurantValidationTest.kt [hunk 6]
```diff
@@ -227,13 +227,13 @@ class RestaurantValidationTest {
...
 
     @Test
     fun `should throw exception when update restaurant with address that not null`() {
         // given a restaurant when address is not null
         val invalidAddressExecutable = Executable {
-            restaurantValidation.validateUpdateRestaurant(fakeRestaurant()[2].copy(address = Address(40.0, 40.0)))
+            restaurantValidation.validateUpdateRestaurantDetails(fakeRestaurant()[2].copy(address = Address(40.0, 40.0)))
         }
         // then check if throw exception
         val error = assertThrows(MultiErrorException::class.java, invalidAddressExecutable)
```
Comment (mustafa-sab3alarab) [inline service_restaurant/src/test/kotlin/org/thechance/service_restaurant/domain/validation/RestaurantValidationTest.kt:239]: not used

**Review by mustafa-sab3alarab — APPROVED** [review]

**Pull request merged**